Add ValuError message for Adamax

Stealeristaken opened the pull request:
Added valueerror messages for adamax optimizer.

awni reviewed (commented):
Thanks! Did you run the pre-commit hooks for formatting? Could you run those and then we can merge?

See the CONTRIBUTING.md doc.

Stealeristaken commented:
> Thanks! Did you run the pre-commit hooks for formatting? Could you run those and then we can merge?
> 
> See the CONTRIBUTING.md doc.

Done testing on a notebook at local. Changed code positions and deleted betas valueerror messages. It works on local. You can merge

awni commented on python/mlx/optimizers.py:
```diff
@@ -423,6 +423,10 @@ def __init__(
         self, learning_rate: float, betas: List[float] = [0.9, 0.999], eps: float = 1e-8
     ):
         super().__init__(learning_rate, betas, eps)
+        if not 0.0 <= learning_rate:
+            raise ValueError(f"Learning rate should be >=0, {self.learning_rate} was provided instead")
```
Is it necessary that the learning rate be `>0`. It seems like a convenient way to do maximization would be to allow positive learning rates. Don't think we should throw in that case

Stealeristaken replied:
I have not come across any studies regarding the use of negative learning rates before. When I look at modern frameworks like TensorFlow and PyTorch, they also assume positive learning rates. Of course, as you suggested, we can do it differently, but I believe we can stick with positive learning rates in MLX as other frameworks do. If there are any studies related to this topic, could you please provide the names? I will make the change accordingly

awni replied:
Yea might it might be safer indeed for this to only use positive learning rates. For SGD it's pretty basic:

`p + lr * g` vs `p - lr * g` does maximization vs minimization.

awni commented:
Thanks!! I think it's good the way it is. I don't feel too strongly about learning rate check either way, but thanks for humoring me 😄

Stealeristaken commented:
> Thanks!! I think it's good the way it is. I don't feel too strongly about learning rate check either way, but thanks for humoring me 😄 

I thank you for the information you provided. I'm relatively new to this, so correcting any mistakes or gaps in my knowledge is beneficial for me. Regards 😁
